checks/cs50/2017/x/finance/checks.py

```diff
@@ -1,13 +1,305 @@
+        self.register("check50user1", "crimson", "crimson", False)
+        self.server_down()
+
+    @check("can_register")
+    def quote_handles_invalid(self):
+        """quote handles invalid ticker symbol."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.quote("ZZZZ", False)
+        self.server_down()
+
+    @check("can_register")
+    def quote_handles_blank(self):
+        """quote handles blank ticker symbol."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.quote("", False)
+        self.server_down()
+
+    @check("can_register")
+    def quote_handles_valid(self):
+        """quote handles valid ticker symbol."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.quote("AAAA", True, value="28.00")
+        self.server_down()
+
+    @check("can_register")
+    def buy_handles_invalid(self):
+        """buy handles invalid ticker symbol."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("buy", "ZZZZ", "2", False)
+        self.server_down()
+
+    @check("can_register")
+    def buy_handles_incorrect_shares(self):
+        """buy handles fractional, negative, and non-numeric shares."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("buy", "AAAA", "-1", False)
+        self.transaction("buy", "AAAA", "1.5", False)
+        self.transaction("buy", "AAAA", "foo", False)
+        self.server_down()
+
+    @check("can_register")
+    def buy_handles_valid(self):
+        """buy handles valid purchase."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("buy", "AAAA", "4", True, value="112.00", remaining="9,?888.00")
+        self.server_down()
+
+    @check("buy_handles_valid")
+    def sell_handles_invalid(self):
+        """sell handles invalid ticker symbol."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("sell", "ZZZZ", "2", False)
+        self.server_down()
+
+    @check("buy_handles_valid")
+    def sell_handles_valid(self):
+        """sell handles valid sell."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("sell", "AAAA", "2", True, value="56.00", remaining="9,?944.00")
+        self.server_down()
+
+    @check("sell_handles_valid")
+    def history(self):
+        """history shows transactions."""
+        self.server_up()
+        self.login("check50user1", "crimson")
+        self.transaction("buy", "BBBB", "1", True)
+        self.driver.get("{}/history".format(Finance.BASE_PATH))
+        if not re.search("14.00", self.driver.page_source):
+            raise Error("Expected to see result of purchase in history.")
+        if not re.search("28.00", self.driver.page_source):
+            raise Error("Expected to see result of sale in history.")
```

Comment:
Errors like this should probably be more explicit, a la "Expected to see result of sale (i.e., 28.00) ..."?